examples/CreatePerTitleEncodingDashAndHls.php

```diff
@@ -0,0 +1,398 @@
+
+    // Wait for the encoding to finish
+    do
+    {
+        $status = $apiClient->encodings()->status($encoding);
+        var_dump(date_create(null, new DateTimeZone('UTC'))->getTimestamp() . ": " . $status->getStatus());
+        $isRunning = !in_array($status->getStatus(), array(Status::ERROR, Status::FINISHED));
+        sleep(1);
+    } while ($isRunning);
+
+    if ($status->getStatus() === Status::ERROR)
+    {
+        var_dump("Encoding failed!");
+        exit();
+    }
+}
+
+/**
+ * This will create and generate Dash and HLS manifests for the given encoding. It will fetch PER_TITLE_RESULT streams
+ * and reference them in the manifests.
+ *
+ * @param ApiClient $apiClient The Bitmovin api-client
+ * @param Encoding $encoding The reference of the encoding
+ * @param Stream audioStream The reference of the audio stream to be linked in the manifests.
+ * @param FMP4Muxing $audiofMp4Muxing The reference of the audio fmp4 muxing for DASH.
+ * @param TSMuxing $audioTSMuxing The reference of the audio TS muxing for HLS.
+ * @throws BitmovinException
+ */
+function generateManifests($apiClient, $encoding, $s3Output, $audioStream, $audiofMp4Muxing, $audioTSMuxing) {
+    // CREATE DASH MANIFEST
+    $manifestOutput = new EncodingOutput($s3Output);
+    $manifestOutput->setOutputPath(OUTPUT_BASE_PATH);
+    $manifestOutput->setAcl(array(new Acl(AclPermission::ACL_PUBLIC_READ)));
+
+    $dashManifest = new DashManifest();
+    $dashManifest->setName("stream.mpd");
+    $dashManifest->setManifestName("stream.mpd");
+    $dashManifest->setOutputs(array($manifestOutput));
+    $dashManifest = $apiClient->manifests()->dash()->create($dashManifest);
+
+    $period = new Period();
+    $period = $apiClient->manifests()->dash()->createPeriod($dashManifest, $period);
+
+    $videoAdaptationSet = new VideoAdaptationSet();
+    $videoAdaptationSet = $apiClient->manifests()->dash()->addVideoAdaptionSetToPeriod($dashManifest, $period, $videoAdaptationSet);
+
+    $audioAdaptationSet = new AudioAdaptationSet();
+    $audioAdaptationSet->setLang('en');
+    $audioAdaptationSet = $apiClient->manifests()->dash()->addAudioAdaptionSetToPeriod($dashManifest, $period, $audioAdaptationSet);
+
+    $audioRepresentation = new DashRepresentation();
+    $audioRepresentation->setType(DashMuxingType::TYPE_TEMPLATE);
+    $audioRepresentation->setEncodingId($encoding->getId());
+    $audioRepresentation->setMuxingId($audiofMp4Muxing->getId());
+    $audioRepresentation->setSegmentPath('audio/dash/128kbps');
+    $apiClient->manifests()->dash()->addRepresentationToAdaptationSet($dashManifest, $period, $audioAdaptationSet, $audioRepresentation);
+
+    $fmp4Muxings = $apiClient->encodings()->muxings($encoding)->fmp4Muxing()->listPage();
+
+    foreach ($fmp4Muxings as $fmp4Muxing) {
+        $segmentPath = $fmp4Muxing->getOutputs()[0]->getOutputPath();
+
+        if (strpos($segmentPath, 'audio') !== false) {
```

Comment:
I would suggest to use a programmatic approach to determine the type of a muxing.

```php 
$streamDetails = $apiClient->encodings()->streams($encoding)->getById($fmp4Muxing->getStreams()[0]->getStreamId());
$codecConfigType = $apiClient->codecConfigurations()->type()->getTypeById($streamDetails->getCodecConfigId());

$streamMode = $streamDetails->getMode();
$isVideoMuxing = in_array($codecConfigType->getType(), array(CodecConfigType::H264, CodecConfigType::H265, CodecConfigType::VP9));
$isAudioMuxing = in_array($codecConfigType->getType(), array(CodecConfigType::AAC));
```